libs/wire-subsystems/src/Wire/ConversationStore/Postgres.hs

```diff
@@ -0,0 +1,1228 @@
+                             WHERE id = ($1 :: uuid)|]
+
+setConversationCellsStateImpl :: (PGConstraints r) => ConvId -> CellsState -> Sem r ()
+setConversationCellsStateImpl convId cells =
+  runStatement (convId, cells) update
+  where
+    update :: Hasql.Statement (ConvId, CellsState) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET cells_state = ($2 :: integer)
+                             WHERE id = ($1 :: uuid)|]
+
+resetConversationImpl :: (PGConstraints r) => ConvId -> GroupId -> Sem r ()
+resetConversationImpl convId groupId =
+  runStatement (convId, groupId) update
+  where
+    update :: Hasql.Statement (ConvId, GroupId) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET group_id = ($2 :: bytea), epoch = 0, epoch_timestamp = NOW()
+                             WHERE id = ($1 :: uuid)|]
+
+setGroupInfoImpl :: (PGConstraints r) => ConvId -> GroupInfoData -> Sem r ()
+setGroupInfoImpl convId groupInfo =
+  runStatement (convId, groupInfo) update
+  where
+    update :: Hasql.Statement (ConvId, GroupInfoData) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET public_group_state = ($2 :: bytea)
+                             WHERE id = ($1 :: uuid)|]
+
+updateChannelAddPermissionsImpl :: (PGConstraints r) => ConvId -> AddPermission -> Sem r ()
+updateChannelAddPermissionsImpl convId addPerm =
+  runStatement (convId, addPerm) update
+  where
+    update :: Hasql.Statement (ConvId, AddPermission) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET channel_add_permission = ($2 :: integer)
+                             WHERE id = ($1 :: uuid)|]
+
+updateToMixedProtocolImpl :: (PGConstraints r) => ConvId -> GroupId -> Epoch -> Sem r ()
+updateToMixedProtocolImpl convId gid epoch =
+  runStatement (convId, ProtocolMixedTag, gid, epoch) update
+  where
+    update :: Hasql.Statement (ConvId, ProtocolTag, GroupId, Epoch) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET protocol = ($2 :: integer), group_id = ($3 :: bytea), epoch = ($4 :: bigint), epoch_timestamp = NOW()
+                             WHERE id = ($1 :: uuid)|]
+
+updateToMLSProtocolImpl :: (PGConstraints r) => ConvId -> Sem r ()
+updateToMLSProtocolImpl convId =
+  runStatement (convId, ProtocolMLSTag) update
+  where
+    update :: Hasql.Statement (ConvId, ProtocolTag) ()
+    update =
+      lmapPG
+        [resultlessStatement|UPDATE conversation
+                             SET protocol = ($2 :: integer), receipt_mode = 0
+                             WHERE id = ($1 :: uuid)|]
+
+-- This doesn't check whether the conv belongs to the team because the cassandra
+-- interpreter doesn't do that either.
```

Comment:
is that a good reason?  i would double-check that it is checked elsewhere, then schedule those checks for removal when cassandra is gone and check it in postgres.

Reply:
We need it in Cassandra world to remove `teamid -> convid` mapping. This gets removed implicitly in postgres.
There is this comment over the Effect constructor:

```
  -- This function only exists to ensure that the cassandra row about team ->
  -- conv relationshop is deleted from cassanrda. This action should be deleted
  -- when we drop support for Cassandra.
```